src/rdf4cpp/datatypes/registry/util/DateTimeUtils.hpp

```diff
@@ -243,29 +243,15 @@ static inline std::partial_ordering compare_time_points(const rdf4cpp::TimePoint
         return cmp(*a, *b);
     };
 
-    if (atz.has_value()) {
-        auto a_sys = apply_timezone(a, *atz);
-        if (btz.has_value()) {
-            return cmp_opt(a_sys, apply_timezone(b, *btz));
-        } else {
-            auto p14 = cmp_opt(a_sys, apply_timezone(b, rdf4cpp::Timezone::max_value()));
-            auto m14 = cmp_opt(a_sys, apply_timezone(b, rdf4cpp::Timezone::min_value()));
-            if (p14 != m14)
-                return std::partial_ordering::unordered;
-            return p14;
-        }
-    } else {
-        if (btz.has_value()) {
-            auto b_sys = apply_timezone(b, *btz);
-            auto p14 = cmp_opt(apply_timezone(a, rdf4cpp::Timezone::max_value()), b_sys);
-            auto m14 = cmp_opt(apply_timezone(a, rdf4cpp::Timezone::min_value()), b_sys);
-            if (p14 != m14)
-                return std::partial_ordering::unordered;
-            return p14;
-        } else {
-            return cmp(a, b);
-        }
+    if (!atz.has_value()) {
```

Comment:
if i understand correctly, if any of the operands do not have a timezone, a default one be applied to them. is that correct?